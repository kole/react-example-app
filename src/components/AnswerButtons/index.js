import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

import Button from '~/components/Button';
import { recordAnswer } from '~/actions/answers';

const AnswerButtonsContainer = styled.div`
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: center;
    min-width: 100%;

    > * {
        display: flex;
        margin: 0 20px;
    }
`;

class Buttons extends React.Component {
    navigateToNextQuestion() {
        const { history, questionID: currentPageNumber, questions } = this.props;

        if (currentPageNumber >= questions.questions.length) {
            return history.push('/results');
        }
        return history.push(`/question/${parseInt(currentPageNumber + 1, 10)}`);
    }

    recordAnswerOnClick(id, answer) {
        const { dispatch } = this.props;
        dispatch(recordAnswer({ id, answer }));
        this.navigateToNextQuestion();
    }

    render() {
        const { questionID: id } = this.props;
        return (
            <AnswerButtonsContainer>
                <Button clickHandler={() => this.recordAnswerOnClick(id, false)} text="False" />
                <Button clickHandler={() => this.recordAnswerOnClick(id, true)} text="True" />
            </AnswerButtonsContainer>
        );
    }
}

Buttons.propTypes = {
    dispatch: PropTypes.func.isRequired,
    questionID: PropTypes.number.isRequired,
    history: PropTypes.object.isRequired,
    questions: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
    answers: state.answers,
    questions: state.questions
});

export default connect(mapStateToProps)(Buttons)
