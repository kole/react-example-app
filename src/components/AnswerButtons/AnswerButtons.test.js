/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import AnswerButtons from './index.js';

const mocks = {
    rootStore: {},
    questionID: 1,
    history: {}
};

describe('AnswerButtons', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(<AnswerButtons history={mocks.history} questionID={mocks.questionID} rootStore={mocks.rootStore} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
