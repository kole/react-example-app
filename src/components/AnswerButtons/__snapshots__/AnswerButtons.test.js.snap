// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerButtons Snapshot 1`] = `
<div
  className="sc-bwzfXH eEgkFF"
>
  <div
    className="sc-bdVaJa bvgpsV"
  >
    <button
      onClick={[Function]}
    >
      False
    </button>
  </div>
  <div
    className="sc-bdVaJa bvgpsV"
  >
    <button
      onClick={[Function]}
    >
      True
    </button>
  </div>
</div>
`;
